code/concepts/concepts.cpp

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+#include <list>
+#include <vector>
+
+/*
+
+In the code below, we provide two implementations of the function increment().
+The second one is optimized for the iterators which fulfill the requirements of
+the concept RandomAccessIterator.
+
+1. Uncomment and complete the concept RandomAccessIterator : check that the class IterT has some operator +=.
+2. Find an existing concept in the standard library, which can replace the previous user-defined concept.
+
+*/
+
+/*
+
+template< typename IterT>
+concept RandomAccessIterator = requires( ... TO BE COMPLETED ... )
```

Comment:
The name `RandomAccessIterator` is not really great here, because being an iterator requires so much more than just `+=`.  You could call it `Advanceable`, since your algorithm essentially does `std::advance`.

Reply:
But it helps the student to find out `std::random_access_iterator` as a possible replacement. What do you think ?